### PR TITLE
Fix for bug when retrieving homonym taxa names

### PR DESCRIPTION
**Pull request by jpjenk**
issue #47

### CODE REVIEW

File: assets/js/navMap.js [hunk 1]
```diff
@@ -1631,12 +1631,19 @@ var navMap = (function() {
...
-    "filterByTaxon": function(name, preventRefresh, isPhylo) {
+    "filterByTaxon": function(name, preventRefresh) {
       if (!name) {
         var name = $("#taxonInput").val();
       }
 
-      d3.json(paleo_nav.dataUrl + paleo_nav.dataService + '/taxa/list.json?name=' + name + '&show=seq', function(err, data) {
+      var selector;
+      
+      if ( name.match(/^txn:|^var:/) )
```
Comment (jczaplew) [inline assets/js/navMap.js]: Although some browsers will be okay with it, don't forget curly brackets with `if else`. For example, 
````
if () {

} else {

}
````

File: assets/js/taxaBrowser.js [hunk 2]
```diff
@@ -16,10 +16,18 @@ var taxaBrowser = (function(){
     if (!name) {
       var name = $("#taxonInput").val();
     }
+
+    var selector;
+      
+    if ( name.match(/^txn:|^var:/) )
```
Comment (jczaplew) [inline assets/js/taxaBrowser.js]: Same thing as above here.

File: assets/js/navMap.js [hunk 3]
```diff
@@ -1631,12 +1631,19 @@ var navMap = (function() {
     },
 
 
-    "filterByTaxon": function(name, preventRefresh, isPhylo) {
+    "filterByTaxon": function(name, preventRefresh) {
```
Comment (jczaplew) [inline assets/js/navMap.js]: I'm not sure what the parameter `isPhylo` did. Is there a particular reason it was removed here?

**Comment by jczaplew** [comment]
Also, be sure to run `gulp` so that the compiled javascript files are created for production deployment.

**Comment by jpjenk** [comment]
@jczaplew, I have added the braces in both locations.  isPhylo simply is not used anywhere in the entire Navigator source code. I would have left it there anyway but I was working with Michael and he just said chuck it.

I ran grunt to build everything before I pushed the branch to GitHub, do I also have to run gulp? If so, I don't have see a task runner script for that.

Having made these corrections, should I now push the branch again?

**Comment by jczaplew** [comment]
@jpjenk My bad - it is definitely `grunt` and not `gulp`. 

Yep! Go ahead and push to this branch.

**Comment by jpjenk** [comment]
@jczaplew I've pushed the new commit on branch homonym_search_fix which just includes the braces. I ran grunt again before pushing.

After you do the merge, who is responsible for pulling the new master to phylum? Also does this need to be merged into the dev branch as well since that is what is running on teststrata? Thanks

**Comment by jczaplew** [comment]
@jpjenk - Looks good. Still not seeing changes to `build/script.min.js` which means `grunt` hasn't been run.

**Comment by jpjenk** [comment]
@jczaplew Hmm.. Just ran it again and tried to commit and push, showing this on my local machine:

=======
➜ navigator git:(homonym_search_fix) ✗ grunt
Running "htmlhint:build" (htmlhint) task
>> 1 file lint free.

Running "uglify:build" (uglify) task
File "build/js/script.min.js" created.

Running "cssmin:combine" (cssmin) task
File build/css/navigator.min.css created.

Done, without errors.
➜  navigator git:(homonym_search_fix) ✗ git commit -a
On branch homonym_search_fix
Untracked files:
	package-lock.json

nothing added to commit but untracked files present
➜  navigator git:(homonym_search_fix) ✗ git push origin homonym_search_fix
Everything up-to-date
➜  navigator git:(homonym_search_fix) ✗

**Comment by jczaplew** [comment]
@jpjenk if you do `git diff` after running `grunt` what do you see?

**Comment by jpjenk** [comment]
@jczaplew, I just checked ./assets/js/navMap.js and taxaBrowser.js and the new {braces} are there. Ran grunt once again and did a git diff ... blank page, it is fully up-to-date with GitHub.  I also ran:

git ls-tree -r homonym_search_fix --name-only

to make sure build/js/script.min.js is included in this branch. It is. Do you think the compiler/minimizer script is just taking the braces back out again anyway since it is a one line if/else? Because, you know, those 32 bits of data are really going to choke the internet :)

**Comment by jczaplew** [comment]
@jpjenk - odd. I just pulled it down and was able to minify the css and js. Not sure, but we should be good now! Have you checked this branch out on dev and tested it?

**Comment by jpjenk** [comment]
@jczaplew I just checked out homonym_search_fix on teststrata but the changes are not there. Now I'm quite confused because git log shows most recent commit is:

commit b7f565793745972bf1a702cf917009cdeb56ddd8
Author: John J Czaplewski <john@czaplewski.org>
Date:   Fri Oct 20 10:29:51 2017 -0500

    Minified css and js

**Comment by jczaplew** [comment]
@jpjenk - could you clarify by what you mean? Are the changes not showing up in the interface?

**Pull request merged**